### PR TITLE
Add more tests of RecordReader Batch Size Edge Cases (#2025)

### PR DESCRIPTION
**Pull request by tustvold**
Follow on from #2027, this just tests a few more edge cases

### CODE REVIEW

File: parquet/src/arrow/arrow_reader.rs [hunk 1]
```diff
@@ -1564,9 +1563,25 @@ mod tests {
...
+        assert_eq!(
+            batch_size,
+            record_reader.next().unwrap().unwrap().num_rows()
+        );
+    }
+
+    #[test]
+    fn test_row_group_exact_multiple() {
+        use crate::arrow::record_reader::MIN_BATCH_SIZE;
+        test_row_group_batch(8, 8);
```
Comment (alamb) [inline parquet/src/arrow/arrow_reader.rs:1580]: 👍 love it

**Review by alamb — APPROVED** [review]
